qemu/tests/driver_load.py

```diff
@@ -94,18 +107,30 @@ def get_driver_id(cmd, pattern):
         session, params["driver_unload_cmd"])
     session.close()
 
-    for repeat in range(0, int(params.get("repeats", 1))):
-        error_context.context("Unload and load the driver. Round %s" % repeat,
-                              logging.info)
-        logging.info("Get driver info from guest")
-        driver_id = get_driver_id(driver_id_cmd, driver_id_pattern)
-
-        error_context.context("Unload the driver", logging.info)
-        unload_driver(driver_unload_cmd, driver_id)
-        time.sleep(5)
-        error_context.context("Load the driver", logging.info)
-        load_driver(driver_load_cmd, driver_id)
-        time.sleep(5)
+    if stop_service_cmd:
+        logging.info("Stop service before driver load testing")
+        service_operate(stop_service_cmd)
+
+    try:
+        for repeat in range(0, int(params.get("repeats", 1))):
+            error_context.context("Unload and load the driver. Round %s" %
+                                  repeat, logging.info)
+            logging.info("Get driver info from guest")
+            driver_id = get_driver_id(driver_id_cmd, driver_id_pattern)
+
+            error_context.context("Unload the driver", logging.info)
+            unload_driver(driver_unload_cmd, driver_id)
+            time.sleep(5)
+            error_context.context("Load the driver", logging.info)
+            load_driver(driver_load_cmd, driver_id)
+            time.sleep(5)
+    finally:
+        if start_service_cmd:
+            logging.info("Restart service after driver load testing")
+            try:
+                service_operate(start_service_cmd)
+            except exceptions.TestError, err:
+                raise exceptions.TestError("%s" % err)
```

Comment:
This is not necessary as service_operate() will raise the TestError.

Reply:
I'm afraid the error from service_operate will cover the exception of driver load/unload, so use a separate try for service_operation